symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/parsing/Arguments.java

```diff
@@ -0,0 +1,82 @@
+package com.symphony.bdk.core.activity.parsing;
+
+import org.apiguardian.api.API;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class storing arguments matching a {@link SlashCommandPattern}
+ */
+@API(status = API.Status.STABLE)
+public class Arguments {
+
+  private Map<String, Object> arguments;
+
+  public Arguments(Map<String, Object> arguments) {
+    this.arguments = arguments;
+  }
+
+  /**
+   *
+   * @return the set of argument names
+   */
+  public Set<String> getArgumentNames() {
+    return arguments.keySet();
+  }
+  /**
+   *
+   * @param argumentName the name of the argument to be retrieved
+   * @return the argument value if present, null otherwise. Actual type of value can be {@link String}, {@link Mention},
+   * {@link Cashtag}, {@link Hashtag} depending on the corresponding {@link SlashCommandPattern}.
+   */
+  public Object get(String argumentName) {
+    return arguments.get(argumentName);
+  }
+
+  /**
+   *
+   * @param argumentName the name of the argument to be retrieved
+   * @return the string representation of the argument if present, null otherwise
+   */
+  public String getAsString(String argumentName) {
+    final Object argumentValue = arguments.get(argumentName);
+    return argumentValue == null ? null : argumentValue.toString();
+  }
+
+  /**
+   *
+   * @param argumentName the name of the argument to be retrieved
+   * @return the argument value if present and if is of type {@link Hashtag}, null otherwise
+   */
+  public Hashtag getAsHashtag(String argumentName) {
```

Comment:
I'd prefer `getHashtag(String argumentName)` instead of `getAsHashtag(String argumentName)`. WDYT @symphony-elias @symphony-youri ?

Reply:
Yes I think we would have a `getString` as well, and potentially `getBoolean` or `getInteger` in the future